idasen/__init__.py

```diff
@@ -75,12 +80,26 @@ def __init__(self, mac: str):
         self._client = BleakClient(self._mac)
 
     async def __aenter__(self):
-        await self._client.__aenter__()
+        await self.connect()
         return self
 
     async def __aexit__(self, *args, **kwargs) -> Optional[bool]:
         return await self._client.__aexit__(*args, **kwargs)
 
+    async def connect(self):
+        i = 0
+        while True:
+            try:
+                await self._client.__aenter__()
+                return
+            except Exception:
+                if i >= self.RETRY_COUNT:
+                    print("Connection failed.", file=sys.stderr)
+                    sys.exit(1)
```

Comment:
So that this can be used as a library if someone does not want to use the CLI:

```suggestion
                    self.logger.warning("Connection failed")
                    raise
```

Reply:
I've added an argument so it will raise an exception instead of calling `sys.exit()`. Calling `sys.exit()` will be, however, the default behavior, as we probably don't want to show tracebacks to the users. Without this argument, we'd have to wrap a lot of code in `try...except` calls to prevent the tracebacks.

Reply:
One last change, could you make the default `exit_on_fail: bool = False`?

`SystemExit` is a subclass of `BaseException`, so this code would not work as most people would expect it to:

```python
try:
    async with IdasenDesk("aa:aa:aa:aa:aa") as desk:
        await desk.is_connected()
except Exception:
    # ...
```

I think `False` by default is a more friendly option for this.

Reply:
I've set it deliberately to `True` to avoid changes in `cli.py`, but I got your point - I've changed it.